tests/tools/importexport/test_reader.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for archive reader."""
+from aiida.tools.importexport.dbimport.readers import ReaderJsonZip
+
+PATH = '/Users/chrisjsewell/Documents/GitHub/aiida_core_develop/out.aiida'
+
+
+def test_json_zip():
+    """Test the JSON zip reader."""
+    with ReaderJsonZip(PATH) as reader:
+        assert reader.metadata.export_version == '0.9'
+        assert reader.metadata.aiida_version == '1.4.2'
+        assert sorted(reader.entity_names) == ['Comment', 'Computer', 'Group', 'Log', 'Node', 'User']
+        assert reader.entity_count('Node') == 40065
+        assert sum(1 for _ in reader.iter_entity_fields('Node')) == 40065
```

Comment:
just wondering - can one not just do something like `len(list(reader.iter_entity_fields('Node')))`?

Reply:
In general the `sum` method here is better for getting the length of an iterator, because with the `len(list` approach you have to read the entire contents of the iterator into memory

Reply:
Thanks for pointing this out, makes sense!